### PR TITLE
Adds GDAS GEMPAK jobs to rocoto task mesh

### LINKED ISSUE #1219: Development workflow lacks GDAS gempak job
**Description**
The development (rocoto) workflow does not include a gdas gempak jobs in the task mesh, so `JGDAS_ATMOS_GEMPAK` and `JGDAS_ATMOS_GEMPAK_META_NCDC` never get exercised.

**Requirements**
Add a GDAS version of the gempak tasks when `DO_GEMPAK` is `YES`.

**Acceptance Criteria (Definition of Done)**
A gfsgempak and a gdasgempak job when gempak jobs are desired.

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
# Description

This PR addresses issue #1219. Rocoto task `jobs/rocoto/gempak.sh` was updated to support `JGDAS_ATMOS_GEMPAK`. Rocoto task `jobs/rocoto/gempakmeta.sh` was added to support `JGFS_ATMOS_GEMPAK_META` and `JGDAS_ATMOS_GEMPAK_META_NCDC`.

  Resolves #1219 
  References #1222 #2073

**NOTE** This is a replicated PR open to resolve developer mishap with Github interface.

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

The Rocoto XML configuration has been executed as follows.

```
user@host$: cd /scratch1/NCEPDEV/da/Henry.Winterbottom/trunk/global-workflow.develop.20231117/workflow
user@host$: ./setup_xml.py /scratch1/NCEPDEV/da/Henry.Winterbottom/work/GLOBAL_WORKFLOW/x002_gfsv17_issue_1219/x002_gfsv17_issue_1219.xml
```

The resulting Rocoto workflow is as follows.

```
       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
202103231200             gfsstage_ic                    51952235              QUEUED                   -         0           0.0
202103231200                 gfsfcst                           -                   -                   -         -             -
202103231200            gfspost_f000                    51952236              QUEUED                   -         0           0.0
202103231200            gfspost_f006                    51952237              QUEUED                   -         0           0.0
202103231200            gfspost_f012                    51952238              QUEUED                   -         0           0.0
202103231200                 gfsvrfy                           -                   -                   -         -             -
202103231200             gfsmetpg2g1                           -                   -                   -         -             -
202103231200             gfsmetpg2o1                           -                   -                   -         -             -
202103231200             gfsmetppcp1                           -                   -                   -         -             -
202103231200               gfsgempak                           -                   -                   -         -             -
202103231200           gfsgempakmeta                           -                   -                   -         -             -
202103231200       gfsgempakmetancdc                           -                   -                   -         -             -
202103231200    gfsgempakncdcupapgif                           -                   -                   -         -             -
202103231200      gfsgempakpgrb2spec                           -                   -                   -         -             -
202103231200                 gfsarch                           -                   -                   -         -             -
202103231200              gfscleanup                           -                   -                   -         -             -

```

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by HenryRWinterbottom** [comment]
@KateFriedman-NOAA @WalterKolczynski-NOAA @aerorahul 

First of all I apologize for requiring multiple PR for this issue. I will try not to make the same errors in the future which have necessitated this.

Secondly, I worked through all of the comments and suggestions from the previous PR attempt (see #2103). If I missed something, please let me know such that I can resolved ASAP.

I appreciate your patience.

File: workflow/applications/gfs_forecast_only.py [hunk 1]
```diff
@@ -129,6 +129,23 @@ def get_task_names(self):
...
+            tasks += ['gempakncdcupapgif']
+            tasks += ['gempakpgrb2spec']
+
+        if self.do_awips:
+            tasks += ['awips_20km_1p0deg']
+            tasks += ['awips_g2']
+            tasks += ['fbwinds']
+
+        if self.do_wafs:
+            tasks += ['wafs', 'wafsgcip', 'wafsgrib2', 'wafsgrib20p25', 'wafsblending', 'wafsblending0p25']
```
Comment (WalterKolczynski-NOAA) [inline workflow/applications/gfs_forecast_only.py:147]: This was moved earlier in the script in #2110 (except WAFS, which was deleted). Please make any changes in the new section and remove this.

File: workflow/applications/gfs_cycled.py [hunk 2]
```diff
@@ -89,9 +89,6 @@ def _get_app_configs(self):
         if self.do_awips:
             configs += ['awips']
 
-        if self.do_npoess:
-            configs += ['npoess']
```
Comment (WalterKolczynski-NOAA) [inline workflow/applications/gfs_cycled.py:93]: Why is this being removed?

File: workflow/applications/gfs_cycled.py [hunk 3]
```diff
@@ -230,12 +231,16 @@ def get_task_names(self):
...
+            gfs_tasks += ['gempakpgrb2spec']
+            gfs_tasks += ['npoess_pgrb2_0p5deg']
 
         if self.do_awips:
             gfs_tasks += ['awips_20km_1p0deg', 'awips_g2', 'fbwinds']
 
-        if self.do_npoess:
-            gfs_tasks += ['npoess']
+        if self.do_wafs:
+            gfs_tasks += ['wafs', 'wafsgcip', 'wafsgrib2', 'wafsgrib20p25', 'wafsblending', 'wafsblending0p25']
```
Comment (WalterKolczynski-NOAA) [inline workflow/applications/gfs_cycled.py:243]: WAFS was removed, and npoess shouldn't have been.
```suggestion
        if self.do_npoess:
            gfs_tasks += ['npoess_pgrb2_0p5deg']
```

File: workflow/applications/applications.py [hunk 4]
```diff
@@ -61,7 +61,6 @@ def __init__(self, conf: Configuration) -> None:
         self.do_genesis = _base.get('DO_GENESIS', True)
         self.do_genesis_fsu = _base.get('DO_GENESIS_FSU', False)
         self.do_metp = _base.get('DO_METP', False)
-        self.do_npoess = _base.get('DO_NPOESS', False)
```
Comment (WalterKolczynski-NOAA) [inline workflow/applications/applications.py:64]: Same, why is this being removed?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: workflow/applications/gfs_cycled.py (hunk 2 above)
Comment (HenryRWinterbottom) [reply workflow/applications/gfs_cycled.py:93]: See https://github.com/NOAA-EMC/global-workflow/pull/2103#discussion_r1415841460

File: workflow/applications/gfs_cycled.py (hunk 3 above)
Comment (HenryRWinterbottom) [reply workflow/applications/gfs_cycled.py:243]: See https://github.com/NOAA-EMC/global-workflow/pull/2103#discussion_r1415841460

File: workflow/applications/applications.py (hunk 4 above)
Comment (HenryRWinterbottom) [reply workflow/applications/applications.py:64]: See https://github.com/NOAA-EMC/global-workflow/pull/2103#discussion_r1415841460

File: workflow/applications/gfs_cycled.py (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply workflow/applications/gfs_cycled.py:93]: Fine doing away with the switch, but you still have to add the config file where the other gempak config is being loaded.

**Pull request closed without merging**